src/veritysetup/veritysetup.c

```diff
@@ -180,9 +245,31 @@ static int run(int argc, char *argv[]) {
                         p.hash_area_offset = arg_hash_offset;
                 }
 
-                r = crypt_load(cd, CRYPT_VERITY, &p);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to load verity superblock: %m");
+                if (!arg_no_superblock) {
+                        p.hash_area_offset = arg_hash_offset;
+
+                        r = crypt_load(cd, CRYPT_VERITY, &p);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to load verity superblock: %m");
+                } else {
+                        p.hash_name = arg_hash;
+                        p.data_device = argv[3];
+                        p.salt = NULL;
+                        p.salt_size = 32;
+                        p.salt = arg_salt;
+                        p.salt_size = arg_salt_size;
+                        p.hash_type = arg_format;
+                        p.data_block_size = arg_data_block_size;
+                        p.hash_block_size = arg_hash_block_size;
+                        p.data_size = arg_data_blocks;
+                        p.hash_area_offset = arg_hash_offset;
+                        p.flags = CRYPT_VERITY_NO_HEADER;
```

Comment:
initialize with structured initialization?

```c
p = (struct crypt_params_verity) {
        …
};
```

Reply:
:facepalm: I have two variables... `params` and `p`.